src/components/batch-selection/batch-selection.stories.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { ComponentStory } from "@storybook/react";
+
+import BatchSelection from ".";
+import IconButton from "../icon-button";
+import Icon from "../icon";
+import Link from "../link";
+
+export const Default = () => (
+  <BatchSelection selectedCount={0}>
+    <span style={{ paddingRight: "5px" }}>
+      (<Link>Select All 38 items</Link>)
+    </span>
+    <IconButton onAction={() => {}}>
+      <Icon type="csv" />
+    </IconButton>
+    <IconButton onAction={() => {}}>
+      <Icon type="bin" />
+    </IconButton>
+    <IconButton onAction={() => {}}>
+      <Icon type="pdf" />
+    </IconButton>
+  </BatchSelection>
+);
+
+export const Themed: ComponentStory<typeof BatchSelection> = (args) => (
+  <BatchSelection {...args}>
+    <IconButton onAction={() => {}}>
+      <Icon type="csv" />
+    </IconButton>
+    <IconButton onAction={() => {}}>
+      <Icon type="bin" />
+    </IconButton>
+    <IconButton onAction={() => {}}>
+      <Icon type="pdf" />
+    </IconButton>
+  </BatchSelection>
+);
+
+export const Dark = Themed.bind({});
+Dark.args = {
+  selectedCount: 1,
+  colorTheme: "dark",
+};
+
+export const Light = Themed.bind({});
+Light.args = {
+  selectedCount: 2,
+  colorTheme: "light",
+};
```

Comment:
I'm not sure why, but some of the `onAction` props in this example are being rendered as `noRefCheck()`. It seems modifying the story to not use `args` fixes the issue? I'm not sure why the first `IconButton`'s `onAction` prop is being displayed correctly though.

![Screenshot 2022-12-12 at 15 31 15](https://user-images.githubusercontent.com/18368713/207085837-12c46bbd-6f0f-4a6c-a9fa-ba387cf9083e.png)

This seems to be the case with the other stories using the `Template.bind({})` pattern as well.

Reply:
That's weird I found an issue in Storybook github repo related to this - https://github.com/storybookjs/storybook/issues/17701

It seems that it has already been fixed, but will be released in v7
https://github.com/storybookjs/storybook/pull/19004

Reply:
Thanks for sharing this @mkrds 👍🏼 Its good to know this has been addressed by the storybook team.

As the release date for Storybook 7.0 doesn't seem to have been announced yet - what do you think about modifying the stories to not use the `Template.bind()` pattern in the meantime?

Reply:
Sure thing @Parsium, fixed :)